src/dlgProfilePreferences.cpp

```diff
@@ -1721,34 +1761,76 @@ void dlgProfilePreferences::copyMap()
 void dlgProfilePreferences::slot_setLogDir()
 {
     Host* pHost = mpHost;
-    if (!pHost)
+    if (!pHost) {
         return;
+    }
 
-    QDir currentLogDir = QFileDialog::getExistingDirectory(this, tr("Select log directory"), pHost->mLogDir, QFileDialog::ShowDirsOnly);
-    mLogDirPath = currentLogDir.absolutePath();
-
-    return;
+    /*
+     * To show the files even though we are looking for a directory so that the
+     * user can see the files that may get appended to depending on the format
+     * selection, we need to use QFileDialog::DontUseNativeDialog because on
+     * Windows the native one does not show files when selecting a directory.
+     *
+     * Also from Qt Docs:
+     * "On Windows, the dialog will spin a blocking modal event loop that will
+     * not dispatch any QTimers, and if parent is not 0 then it will position
+     * the dialog just below the parent's title bar.
+     *
+     * Warning: Do not delete parent during the execution of the dialog. If you
+     * want to do this, you should create the dialog yourself using one of the
+     * QFileDialog constructors."
+     *
+     * That warning suggests *bad things* would happen if the "Save" button or
+     * the widget title bar close button was pressed on the Profile Preferrences
+     * dialog while the directory selector is open...!
+     */
+    // Seems to return "." when Cancel is hit:
+    QDir currentLogDir = QFileDialog::getExistingDirectory(this,
+                                                           tr("Where should Mudlet save log-files?"),
+                                                           (mLogDirPath.isEmpty() ? lineEdit_logFileFolder->placeholderText() : mLogDirPath),
+                                                           QFileDialog::DontUseNativeDialog);
+    if (!(currentLogDir.isRelative() && currentLogDir.path() == QLatin1String("."))) {
+        // This has the side effect of making it not possible to store logs in
+        // the same folder as the executable but we need it to avoid setting the
+        // log file directory to that by accident (clicking on cancel or close)
+
+        if (currentLogDir.absolutePath() != lineEdit_logFileFolder->placeholderText()) {
+            mLogDirPath = currentLogDir.absolutePath();
+            toolButton_resetLogDir->setEnabled(true);
+        } else {
+            // Clear back to show the default as the placehoder text
```

Comment:
When 'Cancel' is clicked, I was thinking that the variable `mLogDirPath` and contents of `lineEdit_logFileFolder` should remain as they were before the file dialog was opened. I realize when a user 'cancels', that usually means the user expects nothing to be changed. Anyway, we have the Reset button if they want to set the user wants to set the log directory to default.